src/components/pages/UploadContacts/index.tsx

```diff
@@ -0,0 +1,259 @@
+import React, { useState, useEffect } from 'react';
+import { RootState } from 'src/redux';
+import Docdrop from 'src/components/docdrop/Docdrop';
+import { connect, ConnectedProps } from 'react-redux';
+import { bindActionCreators, Dispatch } from 'redux';
+import {
+  uploadCsv,
+  removeCsv,
+  updateCsvUploadStep,
+  updateCsvRows,
+  addUploadTag,
+  removeUploadTag,
+  createContacts,
+} from 'src/redux/modules/orgcontacts';
+import FunnelButton from 'src/components/buttons/FunnelButton';
+import ProgressBarHeader from 'src/components/progress/ProgressBarHeader';
+import Image from 'react-bootstrap/Image';
+import Template from 'src/assets/template.png';
+import { readString } from 'react-papaparse';
+import FieldMappingTable from 'src/components/pages/UploadContacts/FieldMappingTable';
+import { initialContactFieldMap } from 'src/data/InitialContactFieldMap';
+import TagSelector from 'src/components/tags/TagSelector';
+import { loadTags, createTag } from 'src/redux/modules/tag';
+import ErrorAlert from 'src/components/alerts/ErrorAlert';
+import { Clock } from 'react-feather';
+import Tag from 'src/components/tags/Tag';
+import { Link } from 'react-router-dom';
+
+const mapStateToProps = (state: RootState) => ({
+  uploadedCsv: state.orgContacts.uploadedCsv,
+  uploadStep: state.orgContacts.uploadStep,
+  tags: state.tags.tags,
+  selectedTags: state.orgContacts.uploadSelectedTags,
+});
+
+const mapDispatchToProps = (dispatch: Dispatch) =>
+  bindActionCreators(
```

Comment:
ohhhh fancy, what exactly does `bindActionCreators` do? (looks cool)

Reply:
Oh hmm, I saw it on a guide and it seemed like a good practise, but maybe it's not the case?
----
Turns an object whose values are action creators, into an object with the same keys, but with every action creator wrapped into a dispatch call so they may be invoked directly.

Normally you should just call dispatch directly on your Store instance. If you use Redux with React, react-redux will provide you with the dispatch function so you can call it directly, too
--- 
How would you do it?